### DISCUSSION TITLE
用skynet实现mmo，每个对象，比如说人物怪物宠物，每个对象一个agent合适吗

### DISCUSSION THREAD

**Discussion by ghost** [cloudwu/skynet#1881] [Q&A]
对象的逻辑都放在场景里，单个场景只能支持400人左右在线。现在想拆分下，不知道合不合适

**Comment by sniper00** [comment]
> 对象的逻辑都放在场景里，单个场景只能支持400人左右在线。现在想拆分下，不知道合不合适

你这样做，逻辑编写难度会大幅度提升，或者参考下星际公民的动态网格技术。其实你可以优先考虑，把耗时的逻辑，用C/Cpp实现，这样做成本是最低的。

**Reply by ghost** [reply]
这工作量也不小啊，用c++重写的话。主要战斗，还有同步那些

**Reply by ghost** [reply]
难度主要会出现在哪些地方

**Reply by sniper00** [reply]
交互和逻辑防重入

**Comment by ghost** [comment]
我想的是利用现在的代码，在场景里做放对象的代理，场景里基本不用改东西，原来对对象的操作，全部由场景里的对象代理转发给对象服务处理。
